tests/kernels/test_marlin_gemm.py

```diff
@@ -231,38 +230,45 @@ def test_gptq_marlin_gemm(
     workspace = MarlinWorkspace(size_n, GPTQ_MARLIN_MIN_THREAD_N,
                                 GPTQ_MARLIN_MAX_PARALLEL)
 
-    opcheck(torch.ops._C.gptq_marlin_gemm,
-            (a_input, marlin_q_w, marlin_s, marlin_zp, g_idx, sort_indices,
-             workspace.scratch, quant_type.id, a_input.shape[0],
-             b_weight.shape[1], a_input.shape[1], is_k_full, False,
-             use_atomic_add, use_fp32_reduce, False),
-            test_utils=DEFAULT_OPCHECK_TEST_UTILS)
-
-    output = ops.gptq_marlin_gemm(
-        a_input,
-        marlin_q_w,
-        marlin_s,
-        marlin_zp,
-        g_idx,
-        sort_indices,
-        workspace.scratch,
-        quant_type,
-        a_input.shape[0],
-        b_weight.shape[1],
-        a_input.shape[1],
-        is_k_full=is_k_full,
-        has_zp=False,
-        use_atomic_add=use_atomic_add,
-        use_fp32_reduce=use_fp32_reduce,
-        is_zp_float=False,
-    )
-    output_ref = torch.matmul(a_input, w_ref)
-
-    torch.cuda.synchronize()
-
-    max_diff = compute_max_diff(output, output_ref)
-
-    assert max_diff < 0.04
+    def check_for_input(a_input):
+        opcheck(torch.ops._C.gptq_marlin_gemm,
+                (a_input, marlin_q_w, marlin_s, marlin_zp, g_idx, sort_indices,
+                 workspace.scratch, quant_type.id, a_input.shape[0],
+                 b_weight.shape[1], a_input.shape[1], is_k_full, False,
+                 use_atomic_add, use_fp32_reduce, False),
+                test_utils=DEFAULT_OPCHECK_TEST_UTILS)
+
+        output = ops.gptq_marlin_gemm(
+            a_input,
+            marlin_q_w,
+            marlin_s,
+            marlin_zp,
+            g_idx,
+            sort_indices,
+            workspace.scratch,
+            quant_type,
+            a_input.shape[0],
+            b_weight.shape[1],
+            a_input.shape[1],
+            is_k_full=is_k_full,
+            has_zp=False,
+            use_atomic_add=use_atomic_add,
+            use_fp32_reduce=use_fp32_reduce,
+            is_zp_float=False,
+        )
+        output_ref = torch.matmul(a_input, w_ref)
+
+        torch.cuda.synchronize()
+
+        max_diff = compute_max_diff(output, output_ref)
+
+        assert max_diff < 0.04
+
+    check_for_input(a_input=rand_data((size_m, size_k)))
+    if not act_order:
```

Comment:
Does this not work for `act_order=True`? Needs to be documented somewhere, with an explanation of why

Reply:
I have added the support for act order.